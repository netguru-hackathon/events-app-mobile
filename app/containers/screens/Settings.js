import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  Image,
  StyleSheet,
  Text,
  View,
} from 'react-native';
import I18n from '../../utils/translations';

const styles = StyleSheet.create({
  icon: {
    width: 20,
    height: 20,
  },
});

class Settings extends Component {
  static propTypes = {
    token: PropTypes.string.isRequired,
  };

  static navigationOptions = {
    tabBarLabel: I18n.t('settings'),
    showIcon: true,
    tabBarIcon: ({ tintColor }) => (
      <Image
        source={require('../../img/SettingsIcon.png')}
        style={[styles.icon, { tintColor }]}
      />
    ),
  };

  render() {
    const { token } = this.props;

    return (
      <View>
        <Text>{I18n.t('settingsTitle')}</Text>
        <Text>{token}</Text>
      </View>
    );
  }
}

const mapStateToProps = state => ({
  token: state.auth.item.token,
});

const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(Settings);
