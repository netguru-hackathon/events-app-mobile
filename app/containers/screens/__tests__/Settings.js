import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Settings from '../Settings';

test('renders correctly', () => {
  const tree = renderer.create(
    <Settings />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
