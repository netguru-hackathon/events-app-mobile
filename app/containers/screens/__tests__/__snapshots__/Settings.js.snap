// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Some Settings text
  </Text>
</View>
`;
